### PR TITLE
Add Content Safety Studio restful APIs

### LINKED ISSUE #23547: [Cognitive Services Responsible AI - Azure Content Safety] API Review
New API Review meeting has been requested.

**Service Name**: Cognitive Services Responsible AI - Azure Content Safety
**Review Created By**: Louise Han
**Review Date**: 05/04/2023 04:00 PM PT
**PR**:  

- https://github.com/Azure/azure-rest-api-specs-pr/pull/12147/files
- https://github.com/Azure/azure-rest-api-specs/pull/23640

**Hero Scenarios Link**: Not Provided
**Core Concepts Doc Link**: https://microsoftapc-my.sharepoint.com/:w:/g/personal/jingruhan_microsoft_com/Eak2-jyV3b9GnKYO00Qei4sBE_dVv4cWvEzCk35ZReutpw?e=mVQ8a2

**Description**: 

Detailed meeting information and documents provided can be accessed [here](https://web.powerapps.com/apps/e41c69de-ef96-44ba-b287-0e012e7c13ae?view=edit&event=0b599ba8-0ada-ed11-a7c7-000d3a341bd5)
For more information that will help prepare you for this review, the requirements, and office hours, visit the documentation [here](https://aka.ms/azsdk/onboarding/restapischedule)

### PR DESCRIPTION
**Pull request by suxi-ms**
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [ ] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc:
* Updated paths:

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

**Comment by suxi-ms** [comment]
Questions:
1. for the query parameter start and end time, which one is suggested to be inclusive?
2.

File: specification/cognitiveservices/ContentSafetyStudio/main.tsp [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+@doc("Error response")
+@error
+model Error {
+  @header("x-ms-error-code")
+  @doc("Error code.")
+  code: int32;
+
+  @doc("Error message.")
+  message: string;
+}
```
Comment (weidongxu-microsoft) [inline specification/cognitiveservices/ContentSafetyStudio/main.tsp:228]: https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#rest-error-response-body-structure

File: specification/cognitiveservices/ContentSafetyStudio/main.tsp [hunk 2]
```diff
@@ -0,0 +1,429 @@
...
+  ...ContentSafetyVersionQueryParameter;
+  ...AggregationQueryParameter;
+}
+
+interface Metrics {
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Does not fit any standard operation pattern"
+  @summary("List API request counts at different timestamps given a time range.")
+  @doc("List API request counts at different timestamps given a time range.")
+  @route("contentSafety/studio/metrics/requestCounts")
+  op listTotalCalls (
```
Comment (weidongxu-microsoft) [inline specification/cognitiveservices/ContentSafetyStudio/main.tsp:340]: Recommend to use operation in Azure.Core, e.g.
https://github.com/Azure/typespec-azure/blob/main/packages/typespec-azure-core/lib/operations.tsp#L18

File: specification/cognitiveservices/ContentSafetyStudio/package.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (tjprescott) [inline specification/cognitiveservices/ContentSafetyStudio/package.json:1]: @suxi-ms this file should be removed and you should rely on the global package.json that is provided at the root of the specs repo.

File: specification/cognitiveservices/ContentSafetyStudio/package-lock.json [hunk 4]
```diff
@@ -0,0 +1,1168 @@
+{
+  "name": "content-safety-studio",
+  "version": "0.1.0",
```
Comment (tjprescott) [inline specification/cognitiveservices/ContentSafetyStudio/package-lock.json:3]: This should also be removed.

**Review by tjprescott — CHANGES_REQUESTED** [review]
Please remove the local package.json files. This should fix the CI that is preventing your APIView from being generated.

File: specification/cognitiveservices/ContentSafetyStudio/package.json (hunk 3 above)
Comment (suxi-ms) [reply specification/cognitiveservices/ContentSafetyStudio/package.json:1]: will do

File: specification/cognitiveservices/ContentSafetyStudio/main.tsp (hunk 2 above)
Comment (suxi-ms) [reply specification/cognitiveservices/ContentSafetyStudio/main.tsp]: updated

File: specification/cognitiveservices/ContentSafetyStudio/main.tsp (hunk 1 above)
Comment (suxi-ms) [reply specification/cognitiveservices/ContentSafetyStudio/main.tsp]: Thanks for feedback， updated

**Review by JeffreyRichter — APPROVED** [review]

File: specification/cognitiveservices/ContentSafetyStudio/main.tsp [hunk 5]
```diff
@@ -0,0 +1,538 @@
...
+}
+
+
+@doc("API request count at a timestamp.")
+@resource("requestCounts")
+model CountPerTimestamp {
+  #suppress "@azure-tools/typespec-azure-core/key-visibility-required" "Properties in response-only schemas should not be marked readOnly"
+  @key("dateTime")
+  @doc("The date time.")
+  dateTime: utcDateTime;
```
Comment (weidongxu-microsoft) [inline specification/cognitiveservices/ContentSafetyStudio/main.tsp:95]: I am not sure if it is best solution to make it `resource` and have `key` on `dateTime`. This `CountPerTimestamp` or `LatencyPerTimestamp` seems not a resource. I assume it only for the `ResourceList` operation?
<-- ignore this comment, if this is recommended from TypeSpec team

Also check [Recommended Naming Conventions](https://github.com/microsoft/api-guidelines/blob/vNext/azure/ConsiderationsForServiceDesign.md#recommended-naming-conventions) -- this is only a recommend.

File: specification/cognitiveservices/ContentSafetyStudio/main.tsp [hunk 6]
```diff
@@ -0,0 +1,538 @@
...
+
+  @doc("The count of the API request.")
+  count: int32;
+}
+
+@doc("Response of calls.")
+@pagedResult
+model CallList {
```
Comment (weidongxu-microsoft) [inline specification/cognitiveservices/ContentSafetyStudio/main.tsp:103]: Delete or comment out if not used.

Use the one from Core, if really need one https://github.com/Azure/typespec-azure/blob/main/packages/typespec-azure-core/lib/foundations.tsp#L231

File: specification/cognitiveservices/ContentSafetyStudio/main.tsp [hunk 7]
```diff
@@ -0,0 +1,538 @@
...
+@doc("Error information that the API returned.")
+@error
+model ErrorResponse {
+
+  @doc("Error code.")
+  code: string;
+
+  @doc("Message that explains the error that the service reported.")
+  message: string;
+}
```
Comment (weidongxu-microsoft) [inline specification/cognitiveservices/ContentSafetyStudio/main.tsp:253]: Delete if not used.

File: specification/cognitiveservices/ContentSafetyStudio/main.tsp [hunk 8]
```diff
@@ -0,0 +1,538 @@
...
+
+
+// Query parameter ////////////////////
+@doc("Provides the 'start' query parameter for list operations. The start time is inclusive.")
+model StartTimeQueryParameter {
+  @query
+  @doc("Start date time inclusive.")
+  start: utcDateTime;
```
Comment (weidongxu-microsoft) [inline specification/cognitiveservices/ContentSafetyStudio/main.tsp:261]: What format does backend accept for the datetime in query parameter?

**Review by tjprescott — APPROVED** [review]

**Pull request closed without merging**